pkg/sfu/buffer/buffer.go

```diff
@@ -1237,6 +1283,40 @@ func (b *Buffer) OnFpsChanged(f func()) {
 	b.Unlock()
 }
 
+func (b *Buffer) OnVideoSizeChanged(fn func([]VideoSize)) {
+	b.Lock()
+	b.onVideoSizeChanged = fn
+	b.Unlock()
+}
+
+// checkVideoSizeChange checks if video size has changed for a specific spatial layer and fires callback
+func (b *Buffer) checkVideoSizeChange(videoSizes []VideoSize) {
+	if len(videoSizes) > len(b.currentVideoSize) {
+		b.logger.Warnw("video size index out of range", nil, "newSize", videoSizes, "currentVideoSize", b.currentVideoSize)
+		return
+	}
+
+	if len(videoSizes) < len(b.currentVideoSize) {
+		videoSizes = append(videoSizes, make([]VideoSize, len(b.currentVideoSize)-len(videoSizes))...)
+	}
```

Comment:
can the number of layers go down? Or is this to handle cases when the number of layers is lower due to congestion?

Reply:
Have the check here becase in simulcast stream it only get 1 videoSize from the media, and svc also could get less than 3 video sizes if the spatial layers is not 3, I align it to the 3 layers with zero size padding to detect the video size change easily. 

> can the number of layers go down

Normally it will not change in the middile, but we can enable svc dynacasting (it is disabled currently for browser compatibility) then it will change.